core/app/models/spree/order.rb

```diff
@@ -168,6 +165,10 @@ def self.register_line_item_comparison_hook(hook)
       line_item_comparison_hooks.add(hook)
     end
 
+    def to_param
+      number.to_s
```

Comment:
@krzysiek1507 It will never return nil or empty string, you cannot create a record with empty number field